File: src/test/java/com/instaimg/crawl/JsonParseTest.java
package com.instaimg.crawl;

import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;
import org.junit.jupiter.api.Test;

public class JsonParseTest {
    @Test
    void name() throws ParseException {
        JSONParser parser = new JSONParser();
        JSONObject jsonObject = (JSONObject) parser.parse("{\n" +
                "    \"id\": \"0001\",\n" +
                "    \"type\": \"donut\",\n" +
                "    \"name\": \"Cake\",\n" +
                "    \"ppu\": 0.55\n" +
                "}");

        System.out.println(jsonObject.containsKey("sasdf"));
    }
}

File: src/test/kotlin/com/instaimg/crawl/JsonParseTest.kt
package com.instaimg.crawl

import org.json.simple.JSONObject
import org.json.simple.parser.JSONParser
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class JsonParseTest {

    private val parser = JSONParser()

    @Test
    fun `containsKey returns true for existing key`() {
        val json = parser.parse("""{"id":"0001","type":"donut"}""") as JSONObject
        assertTrue(json.containsKey("id"))
        assertTrue(json.containsKey("type"))
    }

    @Test
    fun `containsKey returns false for missing key`() {
        val json = parser.parse("""{"id":"0001"}""") as JSONObject
        assertFalse(json.containsKey("nonexistent"))
    }

    @Test
    fun `parse extracts string values correctly`() {
        val json = parser.parse("""{"name":"Cake","type":"donut"}""") as JSONObject
        assertTrue(json["name"] == "Cake")
        assertTrue(json["type"] == "donut")
    }
}
